src/neighborsearch/kball_searcher.jl

```diff
@@ -0,0 +1,68 @@
+# ------------------------------------------------------------------
+# Licensed under the ISC License. See LICENSE in the project root.
+#
+# Author: @exepulveda
+# ------------------------------------------------------------------
+
+"""
+    KBallSearcher(object, k, ball; locations)
+
+A method for searching neighbors in spatial `object` inside `neighborhood`.
+Used when doing estimation combining k-nearest and radius.
+It searches first for the k-nearest by using the internal kdtree and then
+limiting the results within the `BallNeighborhood`
+"""
+struct KBallSearcher{O,K,T,M} <: AbstractBoundedNeighborSearcher
```

Comment:
I think we can simplify the type parameters here. I've learned over time that we sometimes parameterize things too heavily given our C++ background. The idea is to parameterize with the most generic types possible so that the code works with types that we didn't imagine at first:

```julia
struct KBallSearcher{O,B,K} <: AbstractBoundedNeighborSearcher
  object::O
  k::Int
  ball::B
  locations::Vector{Int}
  
  # state fields
  kdtree::K
end
```